src/main/java/org/pac4j/dropwizard/Pac4jConfiguration.java

```diff
@@ -1,7 +1,172 @@
 package org.pac4j.dropwizard;
 
-import io.dropwizard.Configuration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-public interface Pac4jConfiguration<T extends Configuration> {
-    Pac4jFactory getPac4jFactory(T configuration);
+import org.pac4j.core.authorization.authorizer.Authorizer;
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.matching.Matcher;
+import org.pac4j.jax.rs.filter.SecurityFilter;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An interface defining getters for parameters required to configure a {@link SecurityFilter}.
+ */
+public class Pac4jConfiguration {
```

Comment:
Bleh, this is tough. Take a look at dropwizard-db and let me know what you think.

I think a good strategy is for the path of least resistance to be that which is most idiomatically similar to other Dropwizard bundles. This may be, as you say, to target the simplest cases with a Factory class, leaving advanced use cases the ability to drop down to `ConfigSingleton`.

Going forward, we can gauge the level of usage of the factory and readjust our APIs if it ends up not suiting users' needs.